cpp/src/io/comp/unsnap.cu

```diff
@@ -624,24 +624,24 @@ __device__ void snappy_process_symbols(unsnap_state_s* s, int t, Storage& temp_s
  */
 template <int block_size>
 __global__ void __launch_bounds__(block_size)
-  unsnap_kernel(gpu_inflate_input_s* inputs, gpu_inflate_status_s* outputs)
+  unsnap_kernel(device_decompress_input* inputs, decompress_status* outputs)
 {
   __shared__ __align__(16) unsnap_state_s state_g;
   __shared__ cub::WarpReduce<uint32_t>::TempStorage temp_storage;
   int t             = threadIdx.x;
   unsnap_state_s* s = &state_g;
   int strm_id       = blockIdx.x;
 
-  if (t < sizeof(gpu_inflate_input_s) / sizeof(uint32_t)) {
+  if (t < sizeof(device_decompress_input) / sizeof(uint32_t)) {
     reinterpret_cast<uint32_t*>(&s->in)[t] = reinterpret_cast<const uint32_t*>(&inputs[strm_id])[t];
```

Comment:
(This feels like a case where a `reinterpret_cast` has done something very, very unsafe?)

What does `reinterpret_cast<const uint32_t*>(&inputs[strm_id])[t]` mean in this expression? It doesn't refer to the `src` or `dst` members. Is this reinterpreting a `device_span`? The `struct device_decompress_input` itself? It's getting late and my brain is melting so I'm going to wrap up my review here for now.

Reply:
This is a cooperative copy AFAICT. We removed most of these, looks like this one survived. I'll replace with a simple assignment.